src/fu-util.c

```diff
@@ -3676,6 +3676,119 @@ fu_util_show_plugin_warnings(FuUtilPrivate *priv)
 	}
 }
 
+typedef enum {
+	FU_UTIL_DEPENDENCY_KIND_UNKNOWN,
+	FU_UTIL_DEPENDENCY_KIND_RUNTIME,
+	FU_UTIL_DEPENDENCY_KIND_COMPILE,
+} FuUtilDependencyKind;
+
+static const gchar *
+fu_util_dependency_kind_to_string(FuUtilDependencyKind dependency_kind)
+{
+	if (dependency_kind == FU_UTIL_DEPENDENCY_KIND_RUNTIME)
+		return "runtime";
+	if (dependency_kind == FU_UTIL_DEPENDENCY_KIND_COMPILE)
+		return "compile";
+	return NULL;
+}
+
+static gchar *
+fu_util_parse_project_dependency(const gchar *str, FuUtilDependencyKind *dependency_kind)
+{
+	gsize strsz = strlen(str);
```

Comment:
I think you want a g_return_val_if_fail  check that dependency_kind is not a NULL pointer.

Reply:
fixed